### PR TITLE
Simple rotation component.

### LINKED ISSUE #34064: Alt-Click rotation should be a component.
What it says on the tin. 
Should preserve all current wrench/anchored/whatever else checks but making things rotatable should be as easy as adding it.

### PR DESCRIPTION
**Pull request by AnturK**
I'm going to port other rotations to it when i don't feel lazy.
Closes #34064

### CODE REVIEW

File: code/datums/components/rotation.dm [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	var/datum/callback/can_be_rotated  //Check if object can be rotated at all
+	var/datum/callback/after_rotation     //Additional stuff to do after rotation
+	
+	var/rotation_flags = NONE
+
+	//verbs ? 
+	//flipping ?
+
+/datum/component/simple_rotation/Initialize(rotation_flags = NONE ,can_user_rotate,can_be_rotated,after_rotation)
+	src.rotation_flags = rotation_flags
```
Comment (Cyberboss) [inline code/datums/components/rotation.dm]: before this line: 
```dm
if(!ismovableatom(parent))
    return COMPONENT_INCOMPATIBLE
```

**Review by Cyberboss — DISMISSED** [review]

**Comment by coiax** [comment]
Wait, this doesn't go on chairs?

**Comment by AnturK** [comment]
It does, i only converted one example for now.

**Comment by Cyberboss** [comment]
done?

**Comment by AnturK** [comment]
Yes should be good.

File: code/datums/components/rotation.dm [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			rot_degree = -90
+		if(ROTATION_COUNTERCLOCKWISE)
+			rot_degree = 90
+		if(ROTATION_FLIP)
+			rot_degree = 180
+	AM.setDir(turn(AM.dir,rot_degree))
+	after_rotation.Invoke(user,rotation_type)
+
+/datum/component/simple_rotation/proc/default_can_user_rotate(mob/living/user, rotation_type)
+	if(!istype(user) || !user.Adjacent(parent) || user.incapacitated())
```
Comment (ShizCalev) [inline code/datums/components/rotation.dm]: Shouldn't this check user.canmove as well? Otherwise wouldn't pAI be able to rotate things again?

Comment (optimumtact) [reply code/datums/components/rotation.dm]: @AnturK ?

Comment (AnturK) [reply code/datums/components/rotation.dm]: Yes, this is still an issue. pAI's are broken to the point where adding canmove wouldn't really help and proper fix would require full interaction refactor.

**Pull request merged**